src/types/ssh.ts

```diff
@@ -80,6 +80,8 @@ export type SshProvider<
 
   /** Unwraps this provider's types */
   requestToSsh: (request: CliPermissionSpec<PR, O>) => SR;
+  
+  timeLimit: number;
```

Comment:
This should have some jsdoc describing what it does. I might also suggest giving it a more descriptive name:

```suggestion
  /** Amount of time, in ms, to wait between granting access and giving up on attempting an SSH connection */
  propagationTimeoutMs: number;
```

Reply:
Done